lib/spack/spack/package.py

```diff
@@ -1485,6 +1519,152 @@ def _hms(seconds):
+            jobs = 1
+        elif self.make_jobs:
+            jobs = self.make_jobs
+
+        make  = spack.build_environment.MakeExecutable('make', jobs)
+        return make
+
+    def configure_args(self):
+        """Returns package-specific arguments to be provided to the configure command."""
+        return list()
+
+    def configure_env(self):
+        """Returns package-specific environment under which the configure command should be run."""
+        return dict()
+
+    def spack_transitive_include_path(self):
+        return ';'.join(
+            os.path.join(dep, 'include')
+            for dep in os.environ['SPACK_DEPENDENCIES'].split(os.pathsep)
+        )
+
+    def install_setup(self):
+        cmd = [str(which('cmake'))] + \
+            spack.build_environment.get_std_cmake_args(self) + \
+            ['-DCMAKE_INSTALL_PREFIX=%s' % os.environ['SPACK_PREFIX'],
+            '-DCMAKE_C_COMPILER=%s' % os.environ['SPACK_CC'],
+            '-DCMAKE_CXX_COMPILER=%s' % os.environ['SPACK_CXX'],
+            '-DCMAKE_Fortran_COMPILER=%s' % os.environ['SPACK_FC']] + \
+            self.configure_args()
+
+        env = dict()
+        env['PATH'] = os.environ['PATH']
+        env['SPACK_TRANSITIVE_INCLUDE_PATH'] = self.spack_transitive_include_path()
+        env['CMAKE_PREFIX_PATH'] = os.environ['CMAKE_PREFIX_PATH']
+
+        setup_fname = 'spconfig.py'
+        with open(setup_fname, 'w') as fout:
+            fout.write(\
+r"""#!%s
+#
+
+import sys
+import os
+import subprocess
+
+def cmdlist(str):
+	return list(x.strip().replace("'",'') for x in str.split('\n') if x)
+env = dict(os.environ)
+""" % sys.executable)
+
+            env_vars = sorted(list(env.keys()))
+            for name in env_vars:
+                val = env[name]
+                if string.find(name, 'PATH') < 0:
+                    fout.write('env[%s] = %s\n' % (repr(name),repr(val)))
+                else:
+                    if name == 'SPACK_TRANSITIVE_INCLUDE_PATH':
+                        sep = ';'
+                    else:
+                        sep = ':'
+
+                    fout.write('env[%s] = "%s".join(cmdlist("""\n' % (repr(name),sep))
+                    for part in string.split(val, sep):
+                        fout.write('    %s\n' % part)
+                    fout.write('"""))\n')
+
+            fout.write("env['CMAKE_TRANSITIVE_INCLUDE_PATH'] = env['SPACK_TRANSITIVE_INCLUDE_PATH']   # Deprecated\n")
+            fout.write('\ncmd = cmdlist("""\n')
+            fout.write('%s\n' % cmd[0])
+            for arg in cmd[1:]:
+                fout.write('    %s\n' % arg)
+            fout.write('""") + sys.argv[1:]\n')
+            fout.write('\nproc = subprocess.Popen(cmd, env=env)\nproc.wait()\n')
+        make_executable(setup_fname)
+
+
+    def install_configure(self):
+        cmake = which('cmake')
+        with working_dir(self.build_directory, create=True):
+            os.environ.update(self.configure_env())
```

Comment:
Probably no good reason. I used standard python routines because that is
what I knew.
On Jul 8, 2016 6:55 AM, "Massimiliano Culpo" <notifications@github.com>
wrote:

> In lib/spack/spack/package.py:
>
> > +                    fout.write('"""))\n')
> > +
> > +            fout.write("env['CMAKE_TRANSITIVE_INCLUDE_PATH'] = env['SPACK_TRANSITIVE_INCLUDE_PATH']   # Deprecated\n")
> > +            fout.write('\ncmd = cmdlist("""\n')
> > +            fout.write('%s\n' % cmd[0])
> > +            for arg in cmd[1:]:
> > +                fout.write('    %s\n' % arg)
> > +            fout.write('""") + sys.argv[1:]\n')
> > +            fout.write('\nproc = subprocess.Popen(cmd, env=env)\nproc.wait()\n')
> > +        make_executable(setup_fname)
> > +
> > +
> > +    def install_configure(self):
> > +        cmake = which('cmake')
> > +        with working_dir(self.build_directory, create=True):
> > +            os.environ.update(self.configure_env())
>
> @citibeth <https://github.com/citibeth> @tgamblin
> <https://github.com/tgamblin> Is there any reason not to use
> Package.setup_environment and set spack_env there in subclasses of
> CMakePackage ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/LLNL/spack/pull/543/files/6327877a6f9d2c30e4d644ec145a6f3f323ead81#r70057615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AB1cd-hzRHpRBXTqGNT2VUBF7mvo6Jcwks5qTiyXgaJpZM4HvOFN>
> .
>